web3/_utils/request.py

```diff
@@ -23,15 +24,34 @@ def _remove_session(key: str, session: requests.Session) -> None:
 def _get_session(*args: Any, **kwargs: Any) -> requests.Session:
     cache_key = generate_cache_key((args, kwargs))
     if cache_key not in _session_cache:
-        _session_cache[cache_key] = requests.Session()
+        session = requests.Session()
+        adapter = requests.adapters.HTTPAdapter(**extract_adapter_kwargs(**kwargs))
+
+        session.mount('http://', adapter)
+        session.mount('https://', adapter)
+
+        _session_cache[cache_key] = session
     return _session_cache[cache_key]
 
 
 def make_post_request(endpoint_uri: URI, data: bytes, *args: Any, **kwargs: Any) -> bytes:
```

Comment:
If you find this feature useful, it's probably worth adding a small blurb about it in the docs.